symphony/lib/toolkit/class.entrymanager.php

```diff
@@ -391,7 +391,7 @@ public static function fetch($entry_id = null, $section_id = null, $limit = null
 
         $section = SectionManager::fetch($section_id);
         if (!is_object($section)) {
-            return false;
+            return array();
```

Comment:
I find it really surprising that the doc did not mentioned the boolean return. I got caught by this many many times, as you can tell. ;)

Even though, technically, it is not a breaking fix, it think it has the potential of breaking some extensions. We'll see ;)

Reply:
The function's doc block says that the function returns an array. So it should never return `false`, should it? If this breaks an extension, the extension's author hasn't read the docs.

When using the `fetch` method , you should always do `if (EntryManager::fetch())`, never `if (EntryManager::fetch() === false)`.

Reply:
> If this breaks an extension, the extension's author hasn't read the docs.

Ahah I've read the doc multiple time... but when debugging, I'd rather code with what the function is giving me.

`if (array())` returns false ? I've never tried it to be honest...

Reply:
BTW I think other managers have the same problem.

Reply:
Oh yes, it does! See http://php.net/manual/en/language.types.boolean.php#language.types.boolean.casting

This is why `!empty($var)` is the same as `isset($var) && $var`(see http://php.net/manual/en/function.empty.php#refsect1-function.empty-parameters) — and I prefer the latter, because it is more explicit.

Reply:
Heaven or hell: PHP's "type juggling". :-)

Reply:
> BTW I think other managers have the same problem.

So I wouldn't oppose fixing them. I really think that (most) functions should have a single return type.

Reply:
> I prefer the latter, because it is more explicit.

I prefer empty because "it's magic". Thanks Michael!

> Heaven or hell: PHP's "type juggling". :-)

I am still pretty formatted javascript/C/c++/java/.NET even tho I've been using PHP since version 3.

Reply:
> If this breaks an extension, the extension's author hasn't read the docs.

I agree with @michael-e on this. If we had used proper (PHP 7) return types instead of just documenting the return type, the method would have thrown an exception. Extensions should never rely on bugs, accidental or undocumented behavior. If fixing a bug breaks an extension, there’s something wrong with the extension.

Reply:
> Extensions should never rely on bugs, accidental or undocumented behavior.

Indeed. But in the wild, that's not the case. Remember how hidden files in UNIX are actually a bug, that became a feature because people relied on it.

That being said, I do not mind breaking extensions haha. I justed to document that it can be the case (and I'll merge the PR as soon as Jens can do the simple fixes asked in the review).

> Or a nullable return type

Yes I do not consider nullable as a separate  type either.